models/stock_picking_batch.py

```diff
@@ -237,3 +240,82 @@ def is_valid_location_dest_id(self, location_ref):
 
         return all([pick.is_valid_location_dest_id(location=location)
                     for pick in all_done_pickings])
+
+    @api.multi
+    def unpickable_item(self, move_line_id, reason, picking_type_id=None):
+        """
+        Given a valid operation_id create a stock investigation
+        picking for it. If it is the last operation of the wave,
+        the wave is set as done.
+
+        The operation is valid if it is in the current wave (self)
+        and its picking is not done or cancel.
+        """
+        if picking_type_id is None:
+            ResUsers = self.env['res.users']
+            picking_type_id = ResUsers.get_user_warehouse().int_type_id.id
+        self.ensure_one()
+        StockMoveLine = self.env['stock.move.line']
+        Picking = self.env['stock.picking']
+        Groups = self.env['procurement.group']
+        StockQuant = self.env['stock.quant']
+
+        move_line = StockMoveLine.browse(move_line_id)
+        picking = move_line.picking_id
+        package = move_line.package_id
+        location = move_line.location_id
+
+        if not move_line.exists():
+            raise ValidationError(_('Cannot find the operation'))
+        if picking.batch_id != self:
+            raise ValidationError(_('Move line is not part of the batch.'))
+        if picking.state in ['cancel', 'done']:
+            raise ValidationError(_('Cannot mark a move line as unpickable '
+                                    'when it is part of a completed Picking.'))
+
+        if package:
+            # This move line is part of a package.  Get all quants that are
+            # also on this package as well as all other move lines in the
+            # package as they will have been affected as well.
+            quants = package._get_contained_quants()
+            move_lines = picking.move_line_ids.filtered(lambda x:
```

Comment:
We have a function that filters the move_line_ids by package, so you can use something like:
`move_lines = picking.move_line_ids.get_package_move_lines(package)`